src/renderer/features/governance/model/unlock.ts

```diff
@@ -0,0 +1,144 @@
+const flowFinished = createEvent();
+const txSaved = createEvent();
+
+const $step = restore<Step>(stepChanged, Step.NONE);
+const $claimSchedule = createStore<UnlockChunk[]>([]).reset(walletModel.$activeWallet);
+const $isLoading = createStore<boolean>(true);
+
+type Props = {
+  api: ApiPromise;
+  referendums: ReferendumInfo[];
+  tracks: Record<TrackId, TrackInfo>;
+  trackLocks: Record<Address, Record<TrackId, BN>>;
+  voting: VotingMap;
+};
+
+const getClaimScheduleFx = createEffect(
+  async ({ api, referendums, tracks, trackLocks, voting }: Props): Promise<UnlockChunk[]> => {
+    const currentBlockNumber = await getCurrentBlockNumber(api);
+    const undecidingTimeout = api.consts.referenda.undecidingTimeout.toNumber();
+    const voteLockingPeriod = api.consts.convictionVoting.voteLockingPeriod.toNumber();
+
+    const claims = Object.entries(trackLocks).flatMap(([address, trackLock]) => {
+      const claimSchedule = claimScheduleService.estimateClaimSchedule({
+        currentBlockNumber,
+        referendums,
+        tracks,
+        trackLocks: trackLock,
+        votingByTrack: voting[address],
+        voteLockingPeriod,
+        undecidingTimeout,
+      });
+
+      return unlockService.filterClaims(claimSchedule, address);
+    });
+
+    return Promise.all(
+      claims.map((claim) => {
+        if (claim.type !== UnlockChunkType.PENDING_LOCK) return claim;
+
+        return getCreatedDateFromApi((claim.claimableAt as ClaimTimeAt).block, api).then((timeToBlock) => ({
+          ...claim,
+          timeToBlock,
+        }));
+      }),
+    ).then((result) => result);
+  },
+);
+
+// const $unlockStore = createStore<UnlockStore | null>(null);
+
+// const $initiatorWallet = combine(
+//   {
+//     store: $unlockStore,
+//     wallets: walletModel.$wallets,
+//   },
+//   ({ store, wallets }) => {
+//     if (!store) return undefined;
+
+//     return walletUtils.getWalletById(wallets, store.shards[0].walletId);
+//   },
+//   { skipVoid: false },
+// );
+
+sample({
+  clock: [referendumModel.events.requestDone, locksModel.$trackLocks],
+  source: {
+    api: networkSelectorModel.$governanceChainApi,
+    tracks: tracksAggregate.$tracks,
+    trackLocks: locksModel.$trackLocks,
+    voting: votingAggregate.$currentWalletVoting,
+    referendums: referendumModel.$referendums,
+    chain: networkSelectorModel.$governanceChain,
+  },
+  filter: ({ api, voting, trackLocks, chain }) => !!api && !!chain && !!voting && !!trackLocks,
+  fn: ({ api, tracks, trackLocks, voting, referendums, chain }) => ({
+    api: api!,
+    tracks,
+    trackLocks,
+    voting,
+    referendums: referendums[chain!.chainId],
+  }),
+  target: getClaimScheduleFx,
+});
+
+sample({
+  clock: getClaimScheduleFx.doneData,
+  target: $claimSchedule,
+});
+
+sample({
+  clock: flowStarted,
+  fn: () => Step.INIT,
+  target: stepChanged,
+});
+
+sample({
+  clock: getClaimScheduleFx.finally,
+  fn: () => false,
+  target: $isLoading,
```

Comment:
`$isLoading` is true only for first loading, am I right? Because doesn't reset for it. If it should be reseted maybe`getClaimScheduleFx.pending` will work here